### PR TITLE
pass raw errors to  field widgets

### PR DESCRIPTION
**Pull request by llamamoray**
### Reasons for making this change

See issue #727 

The current implementation of ArrayField, BooleanField, and StringField don't pass the error schema down to the widgets they render. This makes rendering errors as part of a custom control tricky. 

This PR solves this by passing the raw errors down, but maintaining the default implementation for displaying errors with the libraries built in widgets.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've checked the rendering of the Markdown text I've added
  - [ ] If I'm adding a new section, I've updated the Table of Content
* [ x] **I'm adding or updating code**
  - [ x] I've added and/or updated tests
  - [x ] I've updated docs if needed
  - [ x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: src/components/widgets/BaseInput.js [hunk 1]
```diff
@@ -22,14 +22,18 @@ function BaseInput(props) {
   const _onChange = ({ target: { value } }) => {
     return props.onChange(value === "" ? options.emptyValue : value);
   };
+
+  const cleanProps = { ...inputProps };
+  delete cleanProps.rawErrors;
```
Comment (glasserc) [inline src/components/widgets/BaseInput.js]: I think you can do `const {rawErrors, ...cleanProps} = inputProps;` See the "Rest in object destructuring section" of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring.

File: src/components/fields/SchemaField.js [hunk 2]
```diff
@@ -113,7 +113,9 @@ function DefaultTemplate(props) {
     <div className={classNames}>
       {displayLabel && <Label label={label} required={required} id={id} />}
       {displayLabel && description ? description : null}
-      {children}
+      {React.Children.map(children, child => {
+        return React.cloneElement(child, { rawErrors: props.rawErrors });
+      })}
```
Comment (glasserc) [inline src/components/fields/SchemaField.js]: This seems a bit heavy. Why is this necessary? Wouldn't it be better to pass the `rawErrors` prop when we constructed widgets?

Comment (llamamoray) [reply src/components/fields/SchemaField.js]: I've just reviewed the code again and I see that this can be passed through the SchemaFieldRender function. I'll update this.

**Comment by llamamoray** [comment]
Thinking about this more, I don't think the ArrayField or ObjectField even need to be passed the raw errors. The objective of the PR is to pass field level errors down to the fields so that custom field components can render errors in any way they wish. I think I can remove the props for ArrayField  and ObjectField.

**Comment by llamamoray** [comment]
@glasserc any feedback on my latest changes?
